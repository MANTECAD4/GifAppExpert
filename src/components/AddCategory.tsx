import { useState } from "react"

type Props = {
    setCategories: React.Dispatch<React.SetStateAction<string[]>>,
}


export const AddCategory = ( {setCategories }: Props) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( event: React.ChangeEvent<HTMLInputElement> ) => {
        setInputValue(event.target.value)
    }

    const onSubmit = ( event: React.FormEvent<HTMLFormElement>  ) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        setCategories(categories => [inputValue,...categories ]);
        setInputValue('');
    }

    return (
        <>
            <form onSubmit={ onSubmit  }>
                <input
                    type="text"
                    placeholder="Search gifs"
                    value={inputValue}
                    onChange={ onInputChange }
                />
            </form>
        </>
    )
}
