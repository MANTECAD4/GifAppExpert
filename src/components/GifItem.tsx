import type { Gif } from '../models/Giphy';
type Props = Gif & {

}

export const GifItem = ( { title, url }:Props ) => {
  return (
    <div className="bg-white m-2 rounded-2xl shadow-md overflow-hidden transition-transform hover:scale-105 cursor-pointer p-2 max-w-xs">
      <img
        src={url}
        alt={title}
        className="w-full h-auto rounded-xl"
      />
      <p className="mt-2 text-center text-sm font-semibold text-gray-700">
        {title}
      </p>
    </div>
  )
}
