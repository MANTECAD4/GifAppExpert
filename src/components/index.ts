
export { AddCategoryToChild } from "./AddCategory/AddCategoryToChild";
export { AddCategoryToFather } from "./AddCategory/AddCategoryToFather";
export { GifGrid } from "./GifGrid";