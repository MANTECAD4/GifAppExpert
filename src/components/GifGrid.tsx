import { useFetchGifs } from '../hooks/useFetchGifs.ts';
import { GifItem } from "./GifItem.tsx";


type Props = {
    category:string
} 


export const GifGrid =  ( { category }:Props ) => {

    const {gifs, isLoading} = useFetchGifs( category );

    return (
        <>
            <h2 key={ category }>{ category }</h2>
            {
                // isLoading 
                // ? (<h2>Cargando</h2>)
                // : (null)
                isLoading && (<p>Loading...</p>)

            }
            <div className="card-grid">
                {
                    gifs.map( gif => (
                        <GifItem 
                            key={ gif.id } 
                            { ...gif }
                        />
                    )) 
                    
                }

            </div>
        </>
  )
}
