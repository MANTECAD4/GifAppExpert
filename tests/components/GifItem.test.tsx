import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';


describe(`Tests on "GifItem.tsx" component`, () => { 
    
    const gifTest = {
        title: "Cat Kittens GIF",
        url: "https://media4.giphy.com/media/v1.Y2lkPTFlMWVjMTVlN2poenVlNmpkdDFkYnA0bTE3MTRmY2k2aXhsMmtzbzl6cmtlZzlvMyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/wAmQh1YWjhGms/200.gif",
        id: "wAmQh1YWjhGms"
    }


    test('The component must match a snapshot', () => { 
        const { container } = render(
            <GifItem 
                key={ gifTest.id }
                { ...gifTest }    
            />
        )  
        expect( container ).toMatchSnapshot();
     });

     test('The image must have the right src and alt  fields', () => { 
        render(
            <GifItem 
                key={ gifTest.id }
                { ...gifTest }    
            />
        );

        const img = screen.getByRole('img');
        if (!(img instanceof HTMLImageElement)) {
            throw new Error('Expected an HTMLImageElement');
        }
        const {src, alt } = img;
        expect(src).toBe(gifTest.url);
        expect(alt).toBe(gifTest.title);

    });

    test('The component must cointain the right title', () => { 

        render(
            <GifItem 
                key={ gifTest.id }
                { ...gifTest }    
            />
        );
        expect(screen.getByText(gifTest.title)).toBeTruthy();
    });

 });