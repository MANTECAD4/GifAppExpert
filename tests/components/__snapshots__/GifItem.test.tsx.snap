// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on "GifItem.tsx" component The component must match a snapshot 1`] = `
<div>
  <div
    class="bg-white m-2 rounded-2xl shadow-md overflow-hidden transition-transform hover:scale-105 cursor-pointer p-2 max-w-xs"
  >
    <img
      alt="Cat Kittens GIF"
      class="w-full h-auto rounded-xl"
      src="https://media4.giphy.com/media/v1.Y2lkPTFlMWVjMTVlN2poenVlNmpkdDFkYnA0bTE3MTRmY2k2aXhsMmtzbzl6cmtlZzlvMyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/wAmQh1YWjhGms/200.gif"
    />
    <p
      class="mt-2 text-center text-sm font-semibold text-gray-700"
    >
      Cat Kittens GIF
    </p>
  </div>
</div>
`;
